src/frontend/src/eth/utils/qr-code.utils.ts

```diff
@@ -19,13 +23,58 @@ export const decodeQrCode = ({
 		return { status: 'cancelled' };
 	}
 
-	const payment = decodePayment(code);
+	const payment = decodeQrCodeUrn(code);
 
 	if (isNullish(payment)) {
 		return { status: 'success', destination: code };
 	}
 
-	const { token, identifier: destination, amount } = payment;
+	const { prefix, destination, amount, address, ethereumChainId, functionName } = payment;
+
+	const normalizeChainId = (chainId: string): string => {
+		if (chainId.startsWith('0x')) {
+			return parseInt(chainId, 16).toString();
+		}
+		return chainId;
+	};
+
+	const matchEthereumToken = (
+		address: string | undefined,
+		functionName: string | undefined,
+		ethereumChainId: string | undefined
+	): string | undefined => {
+		const parsedEthereumChainId = nonNullish(ethereumChainId)
+			? normalizeChainId(ethereumChainId)
+			: 1n;
+		console.warn('parsedEthereumChainId', parsedEthereumChainId);
+
+		if (nonNullish(address) && functionName === 'transfer') {
+			const erc20TokensList = get(erc20Tokens);
+			return (
+				erc20TokensList.find(
+					(token) =>
+						token.address.toLowerCase() === address.toLowerCase() &&
```

Comment:
Not for this PR, just thinking out at loud: we really should create a utils to compare address, I already repeated the `toLowerCase` to many times.

Reply:
Agreed!